### PR TITLE
Support dynamic hits k

### PR DESCRIPTION
**Pull request by pfliu-nlp**
This is a lightweight enhancement, to make the `Hits` metric support dynamic `k`.

### CODE REVIEW

File: explainaboard/analyzers/bar_chart.py [hunk 1]
```diff
@@ -107,17 +107,18 @@ def plot(
...
-            data_mp["x"],
-            data_mp["y"],
-            yerr=[data_mp["y_errormin"], data_mp["y_errormax"]],
-            elinewidth=5,
-            capsize=5,
-        )  # you can use color ="r" for red or skip to default as blue
-        for cap in caps:
-            cap.set_color('gray')
-            cap.set_markeredgewidth(10)
+    # if data_mp["y_errormin"][0] is not None:
```
Comment (neubig) [inline explainaboard/analyzers/bar_chart.py:110]: Delete code instead of commenting.

File: explainaboard/processors/kg_link_tail_prediction.py [hunk 2]
```diff
@@ -111,6 +113,17 @@ def default_features(cls) -> feature.Features:
     def default_metrics(cls) -> list[str]:
         return ["Hits", "MeanReciprocalRank"]
 
+    def _get_metrics(
+        self, sys_info: SysOutputInfo
+    ) -> list[explainaboard.metric.Metric]:
+        config = explainaboard.metric.HitsConfig(
+            hits_k=5
```
Comment (neubig) [inline explainaboard/processors/kg_link_tail_prediction.py:120]: Two comments:

Shall we do the following instead?
`config = HitsConfig(...) if name == 'Hits' else MetricConfig()`

5 is already the default elsewhere, so we probably shouldn't set it here. If it is set here, then even if we change the default in `metric.py` the resulting value would not change, which could be confusing.

**Review by neubig — COMMENTED** [review]
Looks good, I approve with three small suggested changes.

File: explainaboard/processors/kg_link_tail_prediction.py (hunk 2 above)
Comment (pfliu-nlp) [reply explainaboard/processors/kg_link_tail_prediction.py:120]: sounds good!

**Comment by pfliu-nlp** [comment]
Merge this first and will fix one unit test error in another PR.

**Pull request merged**